### PR TITLE
ISPN-11305 Broadcasted queries with projections fail for LAZY iterator

### PR DESCRIPTION
**Pull request by gustavonalle**
https://issues.redhat.com/browse/ISPN-11305

### CODE REVIEW

File: query/src/test/java/org/infinispan/query/blackbox/ClusteredQueryTest.java [hunk 1]
```diff
@@ -446,6 +449,28 @@ public void testBroadcastSortedInfinispanQuery() {
       StaticTestingErrorHandler.assertAllGood(cacheAMachine1, cacheAMachine2);
    }
 
+   @Test
```
Comment (anistor) [inline query/src/test/java/org/infinispan/query/blackbox/ClusteredQueryTest.java]: The ```@Test``` annotation is not needed.

**Comment by anistor** [comment]
I'd merge that last 2 commits

File: query/src/test/java/org/infinispan/query/blackbox/ClusteredQueryTest.java (hunk 1 above)
Comment (gustavonalle) [reply query/src/test/java/org/infinispan/query/blackbox/ClusteredQueryTest.java]: All the other tests are using it, just wanted to follow the pattern

**Comment by gustavonalle** [comment]
Squashed

File: query/src/test/java/org/infinispan/query/blackbox/ClusteredQueryTest.java (hunk 1 above)
Comment (anistor) [reply query/src/test/java/org/infinispan/query/blackbox/ClusteredQueryTest.java]: not in this test class

File: query/src/main/java/org/infinispan/query/clustered/commandworkers/CQLazyFetcher.java [hunk 2]
```diff
@@ -16,8 +19,22 @@
    @Override
    QueryResponse perform(BitSet segments) {
       DocumentExtractor extractor = getQueryBox().get(queryId);
-      Object key = extractKey(extractor, docIndex);
-      Object value = cache.get(key);
+      Object value;
+      try {
+         value = extractValue(extractor, docIndex);
+      } catch (IOException e) {
```
Comment (anistor) [inline query/src/main/java/org/infinispan/query/clustered/commandworkers/CQLazyFetcher.java]: Why not catch the IOE /throw SearchException in the extractValue method. extractKey does it like this already

**Comment by anistor** [comment]
Tests look good . I'll fix the last nitpicking myself and merge this ASAP

**Comment by anistor** [comment]
Merged in master. Thanks @gustavonalle !

**Pull request closed without merging**